src/Widgets/Proxy/ProxyPage.vala

```diff
@@ -53,21 +77,140 @@ namespace Network.Widgets {
+            control_switch.notify["active"].connect (control_switch_activated);
+
+            var control_box = new Gtk.Grid ();
+            control_box.column_spacing = 12;
+            control_box.margin_left = control_box.margin_right = 24;
+            control_box.margin_top = 24;
+            control_box.add (device_img);
+            control_box.add (device_label);
+            control_box.add (control_switch);
+
+#if ENABLE_SYSTEMWIDE_PROXY
+            add (permission_infobar);
+#endif
+            add (control_box);
             add (stackswitcher);
             add (stack);
 
             show_all ();
 
             stack.visible_child = configuration_page;
+
+#if ENABLE_SYSTEMWIDE_PROXY
+            update_infobar_visibility ();
+#endif
+        }
+
+#if ENABLE_SYSTEMWIDE_PROXY
+        private void on_proxy_settings_changed () {
+            if (system_wide_available) {
+                if (proxy_settings.mode == "manual") {
+                    if (http_settings.host != "" && http_settings.port > 0) {
+                        system_proxy_service.set_proxy ("http", "http://%s:%d".printf (http_settings.host, http_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("http", "");
+                    }
+
+                    if (https_settings.host != "" && http_settings.port > 0) {
+                        system_proxy_service.set_proxy ("https", "https://%s:%d".printf (https_settings.host, https_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("https", "");
+                    }
+
+                    if (ftp_settings.host != "" && ftp_settings.port > 0) {
+                        system_proxy_service.set_proxy ("ftp", "ftp://%s:%d".printf (ftp_settings.host, ftp_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("ftp", "");
+                    }
+
+                    if (socks_settings.host != "" && socks_settings.port > 0) {
+                        system_proxy_service.set_proxy ("socks", "socks://%s:%d".printf (socks_settings.host, socks_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("socks", "");
+                    }
+
+                    system_proxy_service.set_no_proxy (string.joinv (",", proxy_settings.ignore_hosts));
+                } else {
+                    system_proxy_service.set_proxy ("http", "");
+                    system_proxy_service.set_proxy ("https", "");
+                    system_proxy_service.set_proxy ("ftp", "");
+                    system_proxy_service.set_proxy ("socks", "");
+                    system_proxy_service.set_no_proxy ("");
+                }
+            }
         }
 
-        protected override void control_switch_activated () {
+        private static Polkit.Permission? get_permission () {
```

Comment:
If this is static, you should have a cached static instance of the permission, so that if this method gets called again (which is not really the case here but anyway) it should return that instance.